packages/@react-native-windows/cli/src/runWindows/utils/deploy.ts

```diff
@@ -44,6 +45,69 @@ export function getBuildConfiguration(options: RunWindowsOptions): BuildConfig {
     : 'Debug';
 }
 
+function shouldDeployByPackage(
+  options: RunWindowsOptions,
+  config: Config,
+): boolean {
+  if (options.deployFromLayout) {
+    // Force deploy by layout
+    return false;
+  }
+
+  let hasAppxSigningEnabled: boolean | null = null;
+  let hasPackageCertificateKeyFile: boolean | null = null;
+
+  // TODO: These two properties should really be determined by
+  // getting the actual values msbuild used during the build,
+  // but for now we'll try to get them manually
+
+  // Check passed in msbuild property overrides
+  if (options.msbuildprops) {
+    const msbuildprops = build.parseMsBuildProps(options);
+    if ('AppxSigningEnabled' in msbuildprops) {
+      hasAppxSigningEnabled =
+        msbuildprops.AppxSigningEnabled.toLowerCase() === 'true';
+    }
+    if ('PackageCertificateKeyFile' in msbuildprops) {
+      hasPackageCertificateKeyFile = true;
+    }
+  }
```

Comment:
should these be augmented with whatever the user passes via `--msbuildprops`?

Reply:
I don't understand what you mean. This code specifically parses `--msbuildprops` and looks for the two props we care about.

Reply:
ah sorry I thought parseMsBuildProps was parsing the project xml to extract those props' values

Reply:
I check what's passed on the command line first as it will override things, and if those properties weren't passed, only then do I spend the time parsing the project file